canvas/api/assignment/fetchsubmissionfiles.py

```diff
@@ -100,25 +102,41 @@ def _write_files(temp_dir, file_contents):
     return len(file_contents)
 
 def _download_files(temp_dir, file_targets):
+    num_files = sum([len(targets) for targets in file_targets.values()])
+    if (num_files == 0):
+        return 0
+
     count = 0
+    max_workers = min(MIN_WORKERS , num_files)
```

Comment:
This seems like way too many workers.
You spent so long in the issue talking about Canvas bucketing system,
but this can easily blow the bucket if there are many files.